tiffslide/tiffslide.py

```diff
@@ -784,7 +784,14 @@ def parse_hamamatsu(self) -> dict[str, Any]:
             "no special hamamatsu-format metadata parsing implemented yet!",
             stacklevel=2,
         )
-        return self.parse_generic_tiff()
+        md = self.parse_generic_tiff()
+        # collect hamamatsu tags
+        page0 = self._tf.series[0][0]
+        hamamatsu_tags = {}
+        for k,t in page0.tags.items():
```

Comment:
Please change this to accessing the relevant Hamamatsu keys specifically, instead of iterating over all tags, and gracefully continue in case a tag is missing.